compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -2422,6 +2415,8 @@ protected ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLo
             case DOUBLE_SLASH_DOUBLE_ASTERISK_LT_TOKEN:
             case SLASH_LT_TOKEN:
                 return ParserRuleContext.XML_NAME_PATTERN;
+            case PARAMETER_WITHOUT_ANNOTS:
+                return ParserRuleContext.TYPE_DESC_IN_PARAM;
```

Comment:
Once we remove above mentioned method. We don't need this recovery rule at all. 
Can remove all usages of `PARAMETER_WITHOUT_ANNOTS` in error handler and remove it from the ParserRuleContex.java as well.